### ISSUE TITLE
Avoid loosing Integration work not already saved as draft when clicking on main menu!

### ISSUE THREAD

**Issue by apupier** [syndesisio/syndesis-ui#1080]
- create an integration project
- provide an initial action
- provide a finish action
- provide some steps
- click on main menu
--> the integration has been lost without warning

**Comment by gashcrumb** [comment]
Yeah, at the very least we need a similar prompt like we do for the connection creation wizard that catches when the user navigates out of the editor.

**Comment by gashcrumb** [comment]
To properly support this we'd need some backend support *or* I guess we could keep the editor state in local storage until the user clicks `Save` or the integration is named.

Or we generate integration names and basically start saving to the backend whenever there's changes.
